### PR TITLE
Improve displayName documentation

### LINKED ISSUE #10438: React.Component Docs - displayName
In https://facebook.github.io/react/docs/react-component.html we have the following statement:

> **displayName**
> The `displayName` string is used in debugging messages. JSX sets this value automatically; see JSX in Depth.

I'm almost sure it's incorrect. Neither React or JSX sets `displayName` automatically (another error: in the docs "JSX in Depth" doesn't even mention `displayName`).

Maybe we could change it to something like:

> **displayName**
> The `displayName` string is used in debugging messages. Usually you don't need to set it explicitly because it's inferred from the function or class name that creates the component. You might want to set it explicitly if you want to display a different name for debugging purposes or when you create a Higher-Order Component, see [Wrap the Display Name for Easy Debugging](https://facebook.github.io/react/docs/higher-order-components.html#convention-wrap-the-display-name-for-easy-debugging) for details.

Makes sense?

### PR DESCRIPTION
**Pull request by hnordt**
Closes #10438

### CODE REVIEW

File: docs/docs/reference-react-component.md [hunk 1]
```diff
@@ -343,7 +343,7 @@ If `props.color` is set to null, it will remain null:
 
 ### `displayName`
 
-The `displayName` string is used in debugging messages. JSX sets this value automatically; see [JSX in Depth](/react/docs/jsx-in-depth.html).
+The `displayName` string is used in debugging messages. Usually you don't need to set it explicitly because it's inferred from the function or class name that creates the component. You might want to set it explicitly if you want to display a different name for debugging purposes or when you create a Higher-Order Component, see [Wrap the Display Name for Easy Debugging](/react/docs/higher-order-components.html#convention-wrap-the-display-name-for-easy-debugging) for details.
```
Comment (aweary) [inline docs/docs/reference-react-component.md]: Let's keep higher-order component lowercased to stay consistent with the rest of the docs.

Comment (hnordt) [inline docs/docs/reference-react-component.md]: @aweary fixed :)

File: docs/docs/reference-react-component.md [hunk 2]
```diff
@@ -343,7 +343,7 @@ If `props.color` is set to null, it will remain null:
 
 ### `displayName`
 
-The `displayName` string is used in debugging messages. JSX sets this value automatically; see [JSX in Depth](/react/docs/jsx-in-depth.html).
+The `displayName` string is used in debugging messages. Usually you don't need to set it explicitly because it's inferred from the function or class name that creates the component. You might want to set it explicitly if you want to display a different name for debugging purposes or when you create a higher-order component, see [Wrap the Display Name for Easy Debugging](/react/docs/higher-order-components.html#convention-wrap-the-display-name-for-easy-debugging) for details.
```
Comment (aweary) [inline docs/docs/reference-react-component.md]: > Usually you don't need to set it explicitly because it's inferred from the function or class name that creates the component.

Can you rephrase this to:

> Usually, you don't need to set it explicitly because it's inferred from the name of the function or class that defines the component.

Comment (hnordt) [inline docs/docs/reference-react-component.md]: @aweary rephased 👍

**Comment by hnordt** [comment]
@aweary hey, is it ready to be merged?

**Comment by sophiebits** [comment]
Thanks @hnordt!
